### DISCUSSION TITLE
How to debug GLFW vulkan support

### DISCUSSION THREAD

**Discussion by adigitoleo** [datoviz/datoviz#48] [Q&A] [answered]
Hi, I'm quite new to cmake et al. and although I think I managed to build datoviz in the past it now has trouble finding the correct GLFW or finding the GLFW Vulkan support. I'm on linux and following the linux installation docs. However after building from HEAD (694b14c92145abbb1c834ba4504487b4f916d14d), all tests until # 034 pass, but that one fails. The same error appears when invoking `./manage.sh demo scatter`:

```
datoviz: /home/void/srcbuilds/datoviz/src/vklite_utils.h:353: backend_extensions: Assertion `(glfwVulkanSupported() != 0)' failed.
./manage.sh: line 198: 31857 Aborted                 VK_INSTANCE_LAYERS=$dump ./build/datoviz test $2
05:11:33.001 I             main.c:0051: running scatter plot demo with 50000 points
datoviz: /home/void/srcbuilds/datoviz/src/vklite_utils.h:353: backend_extensions: Assertion `(glfwVulkanSupported() != 0)' failed.
./manage.sh: line 198: 31932 Aborted                 ./build/datoviz demo
```

The docs says that glfw3 is a "built-in" dependency which I guess means vendored? I can see the `external/imgui/backends/imgui_impl_glfw.h` in the code, that comes from the git submode. I don't exactly follow how this is supposed to end up loaded in `vlkite.h`, where the relevant bit seems to be:

```
#define GFLW_INCLUDE_VULKAN
#include <GLFW/glfw3.h>
```

I suspect this is loading the system GLFW, but I'm not sure why that doesn't work either because `/usr/include/GLFW/glfw3.h` is version 3.3.8 which is the latest release.

I also already had vulkan headers in `/usr/include/vulkan`, now I have slightly newer ones in `/usr/local/include/vulkan`. Tried to rebuild, pointing cmake to the newer one, but no luck.

Is there some kind of cmake incantation required to get things working if I have system libs of vulkan ( + headers ) and glfw(-devel)? Or maybe I'm missing something obvious...

**Comment by rossant** [comment, marked as answer]
I'm not sure what's wrong, it seems like Vulkan [could not find a functional ICD at runtime](https://www.glfw.org/docs/3.3/group__vulkan.html#ga2e7f30931e02464b5bc8d0d4b6f9fe2b). I'd suggest perhaps trying to uninstall vulkan and glfw, start from scratch by installing the [lunarg SDK](https://vulkan.lunarg.com/sdk/home#linux) and try compiling a Vulkan demo, see if that works. Then maybe try a standalone glfw vulkan sample like [this one](https://github.com/glfw/glfw/blob/master/tests/triangle-vulkan.c).
